### PR TITLE
tests: kernel: timer: timer_behavior: Tweak expected std deviation

### LINKED ISSUE #71863: kernel: timer: timer_behavior: kernel.timer.timer fails
**Describe the bug**
The test `tests/kernel/timer/timer_behavior/kernel.timer.timer` fails.

Observed on `nrf5340dk/nrf5340/cpunet`.

**To Reproduce**
1. have `nrf5340dk/nrf5340/cpunet` connected
2. go to your zephyr dir
3. call `./scripts/twister -T tests/kernel/timer/timer_behavior -p nrf5340dk/nrf5340/cpunet --device-testing --device-serial /dev/ttyACM1 -v --inline-logs`
4. See console output with error

**Expected behavior**
Valid console output

**Impact**
Not clear

**Logs and console output**
```
*** Booting Zephyr OS build v3.6.0-2817-gdbf62a5a2a84 ***
Running TESTSUITE timer_jitter_drift
===================================================================
START - test_jitter_drift_timer_period
periodic timer behavior test using built-in restart mechanism
collecting time samples for approx 7 seconds
periodic timer samples gathered, calculating statistics
timer clock rate 32768, kernel tick rate 8192
expected period drift: 98.632812 us
period duration statistics for 7000 samples (0 rollovers):
expected: 1000 us,       32.768000 cycles
min:      1068.115234 us, 35 cycles
max:      1129.150391 us, 37 cycles
mean:     1098.632812 us, 36.000000 cycles
variance: 251.723187 us, 0.270286 cycles
stddev:   15.865787 us, 0.519890 cycles
timer start cycle 12261, end cycle 264261,
total time 7690429.687500 us, expected time 7000000.000000 us,
expected time drift 690429.687500 us, difference 0.000000 us
Assertion failed at ../../../../../../../zephyr/tests/kernel/timer/timer_behavior/src/jitter_drift.c:284: do_test_using: (stddev_us < (double)CONFIG_TIMER_TEST_MAX_STDDEV is false)
Standard deviation (in microseconds) outside expected bound
FAIL - test_jitter_drift_timer_period in 7.858 seconds
===================================================================
START - test_jitter_drift_timer_startdelay
periodic timer behavior test using explicit start with delay
collecting time samples for approx 7 seconds
periodic timer samples gathered, calculating statistics
timer clock rate 32768, kernel tick rate 8192
expected period drift: 98.632812 us
period duration statistics for 7000 samples (0 rollovers):
expected: 1000 us,       32.768000 cycles
min:      1098.632812 us, 36 cycles
max:      1098.632812 us, 36 cycles
mean:     1098.632812 us, 36.000000 cycles
variance: 0.000000 us, 0.000000 cycles
stddev:   0.000000 us, 0.000000 cycles
timer start cycle 270204, end cycle 522204,
total time 7690429.687500 us, expected time 7000000.000000 us,
expected time drift 690429.687500 us, difference 0.000000 us
PASS - test_jitter_drift_timer_startdelay in 7.829 seconds
===================================================================
TESTSUITE timer_jitter_drift failed.
Running TESTSUITE timer_tick_train
===================================================================
START - test_one_tick_timer_train
Initializing 4 Timers, Tick Rate 8192Hz, Expecting 20480 callbacks in 11000 ms
Starting Timers with Skews
Faking busy work, remaining timers is 4, timer callbacks 0
Faking busy work, remaining timers is 4, timer callbacks 684
Faking busy work, remaining timers is 4, timer callbacks 1367
Faking busy work, remaining timers is 4, timer callbacks 2051
Faking busy work, remaining timers is 4, timer callbacks 2734
Faking busy work, remaining timers is 4, timer callbacks 3418
Faking busy work, remaining timers is 4, timer callbacks 4102
Faking busy work, remaining timers is 4, timer callbacks 4785
Faking busy work, remaining timers is 4, timer callbacks 5469
Faking busy work, remaining timers is 4, timer callbacks 6152
Faking busy work, remaining timers is 4, timer callbacks 6836
Faking busy work, remaining timers is 4, timer callbacks 7519
Faking busy work, remaining timers is 4, timer callbacks 8202
Faking busy work, remaining timers is 4, timer callbacks 8886
Faking busy work, remaining timers is 4, timer callbacks 9569
Faking busy work, remaining timers is 4, timer callbacks 10253
Faking busy work, remaining timers is 4, timer callbacks 10937
Faking busy work, remaining timers is 4, timer callbacks 11621
Faking busy work, remaining timers is 4, timer callbacks 12305
Faking busy work, remaining timers is 4, timer callbacks 12989
Faking busy work, remaining timers is 4, timer callbacks 13673
Faking busy work, remaining timers is 4, timer callbacks 14356
Faking busy work, remaining timers is 4, timer callbacks 15040
Faking busy work, remaining timers is 4, timer callbacks 15723
Faking busy work, remaining timers is 4, timer callbacks 16407
Faking busy work, remaining timers is 4, timer callbacks 17090
Faking busy work, remaining timers is 4, timer callbacks 17774
Faking busy work, remaining timers is 4, timer callbacks 18458
Faking busy work, remaining timers is 4, timer callbacks 19142
Faking busy work, remaining timers is 4, timer callbacks 19825
One Tick Timer Train Done, took 10010 ms, busy loop ran 30 times
Perfect delta 16 cycles or 488 us
Timer 0 max delta 17 cycles or 519 us, 0 late callbacks (0.0%)
Timer 1 max delta 16 cycles or 488 us, 0 late callbacks (0.0%)
Timer 2 max delta 17 cycles or 519 us, 0 late callbacks (0.0%)
Timer 3 max delta 16 cycles or 488 us, 0 late callbacks (0.0%)
!! The busy loop didn't run as much as expected.
!! Consider making CONFIG_SYS_CLOCK_TICKS_PER_SEC smaller.
PASS - test_one_tick_timer_train in 10.061 seconds
===================================================================
TESTSUITE timer_tick_train succeeded
------ TESTSUITE SUMMARY START ------
SUITE FAIL -  50.00% [timer_jitter_drift]: pass = 1, fail = 1, skip = 0, total = 2 duration = 15.687 seconds
- FAIL - [timer_jitter_drift.test_jitter_drift_timer_period] duration = 7.858 seconds
- PASS - [timer_jitter_drift.test_jitter_drift_timer_startdelay] duration = 7.829 seconds
SUITE PASS - 100.00% [timer_tick_train]: pass = 1, fail = 0, skip = 0, total = 1 duration = 10.061 seconds
- PASS - [timer_tick_train.test_one_tick_timer_train] duration = 10.061 seconds
------ TESTSUITE SUMMARY END ------
===================================================================
RunID: 6df47e08978b6f7373e505a7011816f1
PROJECT EXECUTION FAILED
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04.1 LTS
 - Toolchain: Zephyr SDk 0.15.6
 - Commit SHA or Version used: `v3.6.0-2817-gdbf62a5a2a84`

### PR DESCRIPTION
**Pull request by nordic-krch**
If frequency of the system clock is lower then deviation may exceed default value (10us). Instead of adjusting the default value, test is rounding up expected standard deviation to a single clock cycle.

Fixes #71863.

### CODE REVIEW

**Comment by teburd** [comment]
I like the change, agree with @golowanow the calculation should be done once before the TC_PRINT and zaasert usage

**Review by andyross — DISMISSED** [review]
Seems reasonable.  FWIW, an alternate technique would be to do the tracking and math in cycle space and not microseconds and just clamp the minimum to "1", which IMHO would be clearer but require more rework.

**Review by npitre — DISMISSED** [review]

**Review by cfriedt — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

File: tests/kernel/timer/timer_behavior/src/jitter_drift.c [hunk 1]
```diff
@@ -283,7 +285,7 @@ static void do_test_using(void (*sample_collection_fn)(void), const char *mechan
 		 ", \"CONFIG_SYS_CLOCK_TICKS_PER_SEC\":%d"
 		 ", \"CONFIG_TIMER_TEST_PERIOD\":%d"
 		 ", \"CONFIG_TIMER_TEST_SAMPLES\":%d"
-		 ", \"CONFIG_TIMER_TEST_MAX_STDDEV\":%d"
+		 ", \"MAX STD DEV\":%d"
```
Comment (golowanow) [inline tests/kernel/timer/timer_behavior/src/jitter_drift.c:288]: same recording output change should be in pytest/test_timer.py as well

**Comment by aescolar** [comment]
@andyross @peter-mitsis please review

File: tests/kernel/timer/timer_behavior/src/jitter_drift.c (hunk 1 above)
Comment (golowanow) [reply tests/kernel/timer/timer_behavior/src/jitter_drift.c:288]: why the record name is different now ?

File: tests/kernel/timer/timer_behavior/pytest/test_timer.py [hunk 2]
```diff
@@ -62,13 +65,13 @@ def do_analysis(test, stats, stats_count, config, sys_clock_hw_cycles_per_sec):
                 f', "min_bound_us":{min_bound * 1_000_000:.6f}'
                 f', "max_bound_us":{max_bound * 1_000_000:.6f}'
                 f', "expected_period_cycles":{expected_period:.0f}'
+                f', "max_stddev":{max_stddev:.6f}'
```
Comment (golowanow) [inline tests/kernel/timer/timer_behavior/pytest/test_timer.py:68]:
```suggestion
                f', "MAX_STD_DEV":{max_stddev:.6f}'
```
record field name should be the same if it is the same value with jitter_drift.c

Comment (nordic-krch) [reply tests/kernel/timer/timer_behavior/pytest/test_timer.py]: done

**Review by golowanow — APPROVED** [review]
LGTM

File: tests/kernel/timer/timer_behavior/pytest/test_timer.py [hunk 3]
```diff
@@ -37,7 +37,10 @@ def do_analysis(test, stats, stats_count, config, sys_clock_hw_cycles_per_sec):
     max_bound = (test_period + period_max_drift * test_period +
                  expected_period_drift) / 1_000_000
 
+    cyc_us = 1000000 / sys_clock_hw_cycles_per_sec
```
Comment (andyross) [inline tests/kernel/timer/timer_behavior/pytest/test_timer.py:40]: Naming whine: many platforms have cycle rates > 1MHz, making this value a zero.   Giving it a name that looks and sounds like "cycles per microsecond" is likely to end up in a divide-by-zero as the code evolves.  It's used as a clamp, so e.g. "`min_cyc = ...; /* Minimum possible cycles that can be measured */`' might be better?

**Review by andyross — APPROVED** [review]
Still looks fine to me.  One nitpick.

**Pull request merged**